plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -176,9 +176,16 @@ func (c *X509Cert) Gather(acc telegraf.Accumulator) error {
 
 			// The first certificate is the leaf/end-entity certificate which needs DNS
 			// name validation against the URL hostname.
-			opts := x509.VerifyOptions{}
+			opts := x509.VerifyOptions{
+				Intermediates: x509.NewCertPool(),
+			}
 			if i == 0 {
 				opts.DNSName = u.Hostname()
+				for j, cert := range certs {
+					if j != 0 {
+						opts.Intermediates.AddCert(cert)
```

Comment:
Don't we need to include these when verifying the other certificates?

Reply:
I'm not certain, it wasn't necessary in the domains I tested. Do you know what a test for that would entail? A self-signed intermediate or something?

Reply:
I found one site that has 2 intermediate certs (4 total in chain) and it didn't need a change for all (valid) certs in chain to be valid.